include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -296,12 +296,18 @@ struct OrtApi {
   OrtStatus*(ORT_API_CALL* EnableMemPattern)(_Inout_ OrtSessionOptions* options)NO_EXCEPTION;
   OrtStatus*(ORT_API_CALL* DisableMemPattern)(_Inout_ OrtSessionOptions* options)NO_EXCEPTION;
 
-  // Enable the memory arena on CPU
+  // Enable a memory arena for CPU allocations.
   // Arena may pre-allocate memory for future usage.
   // set this option to false if you don't want it.
   OrtStatus*(ORT_API_CALL* EnableCpuMemArena)(_Inout_ OrtSessionOptions* options)NO_EXCEPTION;
   OrtStatus*(ORT_API_CALL* DisableCpuMemArena)(_Inout_ OrtSessionOptions* options)NO_EXCEPTION;
 
+  // Enable a memory arena for CUDA allocations.
+  // Arena may pre-allocate memory for future usage.
+  // set this option to false if you don't want it.
+  OrtStatus*(ORT_API_CALL* EnableCudaMemArena)(_Inout_ OrtSessionOptions* options)NO_EXCEPTION;
```

Comment:
Does this mean we need to add more such calls for every EP?

Reply:
I don't think so. AFAIK it's only the CPU and CUDA EPs that use BFCArena currently. If additional EPs start to use it we should have a more generic way of enabling/disabling the arena for each (I think we need a way to specify these sort of finer granularity things without adding a new function or variable to the API each time).